pkgs/tools/misc/mstflint/default.nix

```diff
@@ -4,23 +4,84 @@
 , libibmad
 , openssl
 , zlib
+, xz
+, expat
+, boost
+, curl
+, pkg-config
+, autoreconfHook
+, libxml2
+, pciutils
+, busybox
+, python3
+, minimalBuild ? false
 }:
 
 stdenv.mkDerivation rec {
   pname = "mstflint";
-  version = "4.17.0-1";
+  version = "4.25.0-1";
 
   src = fetchurl {
     url = "https://github.com/Mellanox/mstflint/releases/download/v${version}/mstflint-${version}.tar.gz";
-    sha256 = "030vpiv44sxmjf0dng91ziq1cggwj33yp0l4xc6cdhnrv2prjs7y";
+    hash = "sha256-nYGiWfr8a3q3+bGUb1ovLrAS8/LnEJf+4inIEllW95s=";
   };
 
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+  ];
+
   buildInputs = [
     libibmad
-    openssl
     zlib
+    libxml2
+    openssl
+  ] ++ lib.optional (!minimalBuild) [
+    boost.dev
+    curl.dev
+    expat.dev
+    xz.dev
```

Comment:
Do you really need the dev output of these packages? Shouldn't it be enough to just declare the package name itself?

Reply:
I'll have a look again. When updating this, I got some issues without .dev, but lets recheck :)

Reply:
Without .dev I get issues like this:

```
checking for curl/curl.h... no
configure: error: cannot find curl/curl.h . this header is needed for compiling fw manager tool
error: build of '/nix/store/jmvwp6png1wahnmsfm85kyfdj5wzfkmw-mstflint-4.25.0-1.drv' on 'ssh://root@nixos-dev-01' failed: builder for '/nix/store/jmvwp6png1wahnmsfm85kyfdj5wzfkmw-mstflint-4.25.0-1.drv' failed with exit code 1
error: builder for '/nix/store/jmvwp6png1wahnmsfm85kyfdj5wzfkmw-mstflint-4.25.0-1.drv' failed with exit code 1;
       last 10 log lines:
       > ./configure: line 17095: ofed_info: command not found
       > ./configure: line 17096: ofed_info: command not found
       > ./configure: line 17105: test: 5.6: unary operator expected
       > checking for termios.h... yes
       > checking --enable-dc argument... yes
       > checking for zlib.h... yes
       > checking for uncompress in -lz... yes
       > checking --enable-fw-mgr argument... yes
       > checking for curl/curl.h... no
       > configure: error: cannot find curl/curl.h . this header is needed for compiling fw manager tool
       For full logs, run 'nix log /nix/store/jmvwp6png1wahnmsfm85kyfdj5wzfkmw-mstflint-4.25.0-1.drv'.
  ```

Reply:
Weird, I would expect splicing to happen here. Maybe @Artturin can chime in here.

Reply:
> Weird, I would expect splicing to happen here. Maybe @Artturin can chime in here.

Splicing is not relevant here (or in any non-cross case currently)

The dev output selection is done in stdenv https://github.com/NixOS/nixpkgs/blob/ed605168c0b9529d6f052e8fb3fef3607db4f879/pkgs/stdenv/generic/make-derivation.nix#L249

The issue here is that `lib.optional` is used instead of `lib.optionals`

```
nix-repl> mstflint.buildInputs
[ «derivation /nix/store/y2ckb58lw8lpy9cgidlvqx7yhsk6l2ga-libibmad-1.3.13.drv» «derivation /nix/store/mlg49kdfzri70jjl8vnkk8kjfjpf9yam-zlib-1.3.drv» «derivation /nix/store/lblpmc78zda6xf0pnkd534x1lsgaqcxy-libxml2-2.11.5.drv» «derivation /nix/store/f6ry6jbzhfbgqq3hs9fl9b4df7qjyy23-openssl-3.0.11.drv» [ ... ] ]
```

vs
```
nix-repl> mstflint.buildInputs
[ «derivation /nix/store/y2ckb58lw8lpy9cgidlvqx7yhsk6l2ga-libibmad-1.3.13.drv» «derivation /nix/store/mlg49kdfzri70jjl8vnkk8kjfjpf9yam-zlib-1.3.drv» «derivation /nix/store/lblpmc78zda6xf0pnkd534x1lsgaqcxy-libxml2-2.11.5.drv» «derivation /nix/store/f6ry6jbzhfbgqq3hs9fl9b4df7qjyy23-openssl-3.0.11.drv» «derivation /nix/store/mpnldpfhn7sbr1rrz8kpn25d72v62fvk-boost-1.81.0.drv» «derivation /nix/store/4hjpj9n6jlvi2wi6w5wnh997r9qhf912-curl-8.4.0.drv» «derivation /nix/store/6jmg7rbfv6hvpszjv69i01dkbz9y6rrv-expat-2.5.0.drv» «derivation /nix/store/arapifn6rarkgs04zxkpffv7q2yf55vl-xz-5.4.4.drv» «derivation /nix/store/45cxblcr9hhi2k75y8m0mr80y5vbkpj1-python3-3.11.5.drv» ]
```

Reply:
Thanks a lot for this clarification. Now it works like a charm without .dev :+1: